pyclam/manifold.py

```diff
@@ -202,36 +202,37 @@ def argmax_max(b):
                 return b[argmax], distances[argmax]
 
             argradii_radii = [argmax_max(batch) for batch in iter(self)]
-            _argradius, _radius = max(argradii_radii, key=itemgetter(1))
-            self.__dict__['_argradius'], self.__dict__['_radius'] = int(_argradius), float(_radius)
-        return self.__dict__['_argradius']
+            argradius, radius = max(argradii_radii, key=itemgetter(1))
+            self.cache['argradius'], self.cache['radius'] = int(argradius), float(radius)
+        return self.cache['argradius']
 
     @property
     def local_fractal_dimension(self) -> float:
         """ The local fractal dimension of the cluster. """
-        if '_local_fractal_dimension' not in self.__dict__:
+        if 'local_fractal_dimension' not in self.cache:
             logging.debug(f'building cache for {self}')
             if self.nsamples == 1:
                 return 0.
             count = [d <= (self.radius / 2)
                      for batch in iter(self)
                      for d in self.distance_from(batch)]
             count = np.sum(count)
-            self.__dict__['_local_fractal_dimension'] = count if count == 0. else np.log2(len(self.argpoints) / count)
-        return self.__dict__['_local_fractal_dimension']
+            self.cache['local_fractal_dimension'] = count if count == 0. else np.log2(len(self.argpoints) / count)
+        return self.cache['local_fractal_dimension']
 
     @property
     def optimal(self) -> bool:
-        return self.__dict__['_optimal']
+        return self.cache['optimal']
+
+    @property
+    def absorbable(self) -> bool:
+        return self.cache['absorbable']
 
     def clear_cache(self) -> None:
         """ Clears the cache for the cluster. """
         logging.debug(f'clearing cache for {self}')
-        for prop in ['_argsamples', '_argmedoid', '_argradius', '_radius', '_local_fractal_dimension', '_optimal']:
-            try:
-                del self.__dict__[prop]
-            except KeyError:
-                pass
+        self.cache = {'optimal': False, 'absorbable': False}
```

Comment:
Seems weird to set values in clear

Reply:
resets to defaults. See the init method.
In any case, I may have figured out how to not have those two flags in Cluster anyway